### PR TITLE
Drugsyn benchmark group

### PR DESCRIPTION
**Pull request by yhr91**
There are two splits:

- one is by metric (css, loewe, zip, bliss, hsa)
- the other is by tissue type (cell lines can be grouped into 9 tissues

Also added code to the base benchmark class for dealing with pickle files instead of csv

### CODE REVIEW

**Comment by yhr91** [comment]
**Table 1** 
Metric | Number | Task | Metric | Split
-- | -- | -- | -- | -- 
CSS | 297098 | Regression | MAE | Random
Loewe | 297098 | Regression | MAE | Random
Zip | 297098 | Regression | MAE | Random
Bliss | 297098 | Regression | MAE | Random
HSA | 297098 | Regression | MAE | Random


**Table 2**
Tissue | Metric | Number | Task | Metric | Split
-- | -- | -- | -- | -- | --
Kidney | CSS | 40108 | Regression | MAE | Random
Lung | CSS | 45753 | Regression | MAE | Random
Breast | CSS | 20551 | Regression | MAE | Random
Hematopoietic and Lymphoid | CSS | 29224 | Regression | MAE | Random
Colon | CSS | 36624 | Regression | MAE | Random
Prostate | CSS | 10063 | Regression | MAE | Random
Ovary | CSS | 36153 | Regression | MAE | Random
Skin | CSS | 43701 | Regression | MAE | Random
Breast | CSS | 5068 | Regression | MAE | Random
Brain | CSS | 29853 | Regression | MAE | Random

**Comment by yhr91** [comment]
I've updated Table 2. The evaluator now simply runs additional comparisons on the test set for each tissue so we don't use separate training sets for each tissue. [See updates blow]

**Table 1 Predicting drug-drug interactions and combination sensitivity across all tissues**

- The drug combination sensitivity scores (CSS) is determined as the average area under curve (AUC) for the combinations’ dose–response with one compound fixed at the IC50 concentration. 
- The degree of drug-drug interactions can be assessed over the full dose–response matrix using the expected effect of non-interaction. Four commonly-used reference models for this are Bliss independence (BLISS), Highest single agent (HSA), Loewe additivity (LOEWE), and Zero interaction potency (ZIP)


Metric | Number | Task | Metric | Split
-- | -- | -- | -- | -- 
CSS | 297098 | Regression | MAE | Random
Loewe | 297098 | Regression | MAE | Random
Zip | 297098 | Regression | MAE | Random
Bliss | 297098 | Regression | MAE | Random
HSA | 297098 | Regression | MAE | Random

The training dataset was designed such that none of the drug combinations in the test set were present in the training set. Each combination in the test set was used to treat each cell line in the dataset. The cell lines can be classified by organ to measure performance across different organs.

**Table 2 Predicting combination sensitivity within specific tissues**
Tissue | Metric | Number of cell lines | Test set Size | Task | Metric | Split
-- | -- | -- | -- | -- | -- | --
Kidney | CSS | 8 | 8096 | Regression | MAE | Random
Lung | CSS | 9 | 9108 | Regression | MAE | Random
Breast | CSS | 5 | 20551 | Regression | MAE | Random
Hematopoietic and Lymphoid | CSS | 6 | 6072 | Regression | MAE | Random
Colon | CSS | 7 | 7084 | Regression | MAE | Random
Prostate | CSS | 2 | 2024 | Regression | MAE | Random
Ovary | CSS | 7 | 7084 | Regression | MAE | Random
Skin | CSS | 9 | 9108 | Regression | MAE | Random
Brain | CSS | 6 | 6072 | Regression | MAE | Random

There is no separate data download for the tissue breakdown. I've added an additional column to test.pkl to handle the additional metrics

Sample script
```
group = BenchmarkGroup(name = 'drugsyn_group', path = 'data/',
                       file_format='pkl')

predictions = {}

for benchmark in group:
    name = benchmark['name']
    train, valid, test = benchmark['train'], benchmark['valid'], benchmark['test']

    ## --- train your model --- ##

    predictions[name] = pred

out = group.evaluate(predictions)

print(out)
"{'drugcomb_css': {'mae': 22.963}, 'drugcomb_css_kidney': {'mae': 21.811}, 'drugcomb_css_lung': {'mae': 21.367},
'drugcomb_css_breast': {'mae': 18.25}, 'drugcomb_css_hematopoietic_lymphoid': {'mae': 40.379}, 'drugcomb_css_colon': {'mae': 25.001},
'drugcomb_css_prostate': {'mae': 21.874}, 'drugcomb_css_ovary': {'mae': 19.468}, 'drugcomb_css_skin': {'mae': 18.683}, 
'drugcomb_css_brain': {'mae': 21.89}, 'drugcomb_hsa': {'mae': 4.562}, 'drugcomb_loewe': {'mae': 11.109}, 'drugcomb_bliss': {'mae': 4.686}, 
'drugcomb_zip': {'mae': 4.545}}
"
```

**Comment by yhr91** [comment]
**Table 1 Predicting drug-drug interactions and combination sensitivity across all tissues**

- The drug combination sensitivity scores (CSS) is determined as the average area under curve (AUC) for the combinations’ dose–response with one compound fixed at the IC50 concentration. 
- The degree of drug-drug interactions can be assessed over the full dose–response matrix using the expected effect of non-interaction. Four commonly-used reference models for this are Bliss independence (BLISS), Highest single agent (HSA), Loewe additivity (LOEWE), and Zero interaction potency (ZIP)


Metric | Number | Task | Metric | Split
-- | -- | -- | -- | -- 
CSS | 297098 | Regression | MAE | Combination
Loewe | 297098 | Regression | MAE | Combination
Zip | 297098 | Regression | MAE | Combination
Bliss | 297098 | Regression | MAE | Combination
HSA | 297098 | Regression | MAE | Combination


**Table 2 Predicting combination sensitivity within specific tissues**
Tissue | Metric | Number of cell lines | Test set Size | Task | Metric | Split
-- | -- | -- | -- | -- | -- | --
Kidney | CSS | 8 | 8096 | Regression | MAE | Combination
Lung | CSS | 9 | 9108 | Regression | MAE | Combination
Breast | CSS | 5 | 20551 | Regression | MAE | Combination
Hematopoietic and Lymphoid | CSS | 6 | 6072 | Regression | MAE | Combination
Colon | CSS | 7 | 7084 | Regression | MAE | Combination
Prostate | CSS | 2 | 2024 | Regression | MAE | Combination
Ovary | CSS | 7 | 7084 | Regression | MAE | Combination
Skin | CSS | 9 | 9108 | Regression | MAE | Combination
Brain | CSS | 6 | 6072 | Regression | MAE | Combination



Sample script
```
group = BenchmarkGroup(name = 'drugsyn_group', path = 'data/',
                       file_format='pkl')

predictions = {}

for benchmark in group:
    name = benchmark['name']
    train, valid, test = benchmark['train'], benchmark['valid'], benchmark['test']

    ## --- train your model --- ##

    predictions[name] = pred

out = group.evaluate(predictions)

print(out)
"{'drugcomb_css': {'mae': 22.963}, 'drugcomb_css_kidney': {'mae': 21.811}, 'drugcomb_css_lung': {'mae': 21.367},
'drugcomb_css_breast': {'mae': 18.25}, 'drugcomb_css_hematopoietic_lymphoid': {'mae': 40.379}, 'drugcomb_css_colon': {'mae': 25.001},
'drugcomb_css_prostate': {'mae': 21.874}, 'drugcomb_css_ovary': {'mae': 19.468}, 'drugcomb_css_skin': {'mae': 18.683}, 
'drugcomb_css_brain': {'mae': 21.89}, 'drugcomb_hsa': {'mae': 4.562}, 'drugcomb_loewe': {'mae': 11.109}, 'drugcomb_bliss': {'mae': 4.686}, 
'drugcomb_zip': {'mae': 4.545}}
```

For auxiliary training and validation splits:
```
out = group.get_auxiliary_train_valid_split(seed = 42, benchmark = 'drugsyn_css')
train, valid = out['train'], out['valid']
## --- train your model --- ##
group.evaluate(y_pred_val, y_true_val, benchmark = 'drugsyn_css')
```

**Review by kexinhuang12345 — APPROVED** [review]
